app/lib/blog.server.ts

```diff
@@ -84,14 +85,8 @@ function getAuthor(name: string): BlogAuthor | undefined {
   return AUTHORS.find((a) => a.name === name);
 }
 
-function getValidAuthorNames(authors: string[]) {
-  let validatedAuthors: string[] = [];
-  for (let author of AUTHORS) {
-    if (authors.includes(author.name)) {
-      validatedAuthors.push(author.name);
```

Comment:
The alphabetical order was a side effect of iterating over the complete author list rather than filtering the input author list.